site/cds_rdm/components.py

```diff
@@ -50,6 +51,19 @@ def is_thesis(record):
 class CDSResourcePublication(ServiceComponent):
     """CDS resource publication component."""
 
+    def _validate_programme_participation(self, identity, record_or_draft, errors=None):
+        """Validate that user has selected a programme participation."""
+        if record_or_draft.get("custom_fields", {}).get("cern:programmes") is None:
+
+            error_message = _(
+                f"Select the student programme in which you have participated."
+            )
+            if errors is None:
+                errors = []
```

Comment:
we should raise here no? Otherwise, if via REST API someone publishes directly without saving first then the validation will not kick in...